node/api/api.go

```diff
@@ -162,3 +184,77 @@ func WriteJSON(w http.ResponseWriter, obj interface{}) {
 func WriteSuccess(w http.ResponseWriter) {
 	w.WriteHeader(http.StatusNoContent)
 }
+
+func GetPaginationDefaultRequest(req *http.Request) PaginationRequest {
+	return GetPaginationRequest(req, "start", "limit")
+}
+
+func GetPaginationRequest(req *http.Request, startQueryParam string, limitQueryParam string) PaginationRequest {
+	startString := req.FormValue(startQueryParam)
+	if startString == "" {
+		return PaginationRequest{
+			HasPaginationQuery:     false,
+			PaginationQueryIsValid: false,
+			Start: 0,
+			Limit: 0,
+		}
+	}
+	startIndex, err := strconv.Atoi(startString)
+	if err != nil || startIndex < 0 {
+		return PaginationRequest{
+			HasPaginationQuery:     true,
+			PaginationQueryIsValid: false,
+			Start: 0,
+			Limit: 0,
+		}
+	}
+	limit := DefaultPaginationSize
+	limitString := req.FormValue(limitQueryParam)
+	if limitString != "" {
+		limit, err = strconv.Atoi(limitString)
+		if err != nil || limit <= 0 {
+			return PaginationRequest{
+				HasPaginationQuery:     true,
+				PaginationQueryIsValid: false,
+				Start: 0,
+				Limit: 0,
+			}
+		}
+	}
+	return PaginationRequest{
+		HasPaginationQuery:     true,
+		PaginationQueryIsValid: true,
+		Start: startIndex,
+		Limit: limit,
+	}
+}
+
+// returns the starting index, end index, and total pages of a given slice for the page
+func GetPaginationIndicesAndResponse(sliceInterface interface{}, paginationRequest PaginationRequest) (PaginationWrapper, PaginationResponse) {
+	if paginationRequest.Limit <= 0 {
+		build.Critical("pagination request limit must be greater than 0")
+	}
+	slice := reflect.ValueOf(sliceInterface)
+	if slice.Kind() != reflect.Slice {
+		build.Critical("attempting to paginate on non-slice object type: ", slice.Kind())
+	}
+	startingPageIndex := int(math.Min(float64(paginationRequest.Start), float64(slice.Len())))
+	endingPageIndex := int(math.Min(float64(paginationRequest.Start+paginationRequest.Limit), float64(slice.Len())))
+	totalPages := slice.Len() / paginationRequest.Limit
+	if math.Mod(float64(slice.Len()), float64(paginationRequest.Limit)) != 0 {
+		totalPages += 1
+	}
+
+	pagination := PaginationWrapper{
+		Start:      startingPageIndex,
+		End:        endingPageIndex,
+		TotalPages: totalPages,
+	}
+
+	paginationResponse := PaginationResponse{
+		Start:      startingPageIndex,
+		Limit:      paginationRequest.Limit,
+		TotalPages: totalPages,
+	}
+	return pagination, paginationResponse
```

Comment:
`pagination` and `paginationResponse` have exactly the same fields. Why not just do a 
`return startPageIndex, endingPageIndex, totalPages? That reduces the length of the method by 50%.
You might even want to use named return values instead. That way it is just a `return` at the end of the function.